src/env0-environments-provider.ts

```diff
@@ -41,19 +41,15 @@ export class Environment extends vscode.TreeItem {
 export class Env0EnvironmentsProvider
   implements vscode.TreeDataProvider<Environment>
 {
-  private environments: Environment[];
-
-  constructor() {
-    this.environments = [];
-  }
+  private environmentTreeItems: Environment[] = [];
+  private environments: EnvironmentModel[] = [];
```

Comment:
why the separation? it used to only be `environments`. why do you need both?

Reply:
since i moved the loading of environments from `getChildren` to the refresh function i saved added them as class member to use them in `getChildren`.
i can map the environments to tree items in the refresh function and save only the tree items, do you think its a better solution?

Reply:
im sorry, I don't understand

Reply:
Forget about that 😉.
fixed 8462044354a2a51b4fa88a82b2a62ad8f6afdb01